cpp/src/replace/replace.cu

```diff
@@ -13,26 +16,551 @@
+ *        `replace_nulls` with the appropriate data types.
+ */
+/* ----------------------------------------------------------------------------*/
+struct replace_nulls_scalar_kernel_forwarder {
+  template <typename col_type>
+  void operator()(cudf::column_view const& input,
+                  cudf::scalar const* replacement,
+                  cudf::mutable_column_view& output,
+                  cudaStream_t stream = 0,
+                  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource())
+  {
+    cudf::size_type nrows = input.size();
+    cudf::util::cuda::grid_config_1d grid{nrows, BLOCK_SIZE};
+
+    using ScalarType = cudf::experimental::scalar_type_t<col_type>;
+    auto s1 = static_cast<ScalarType const*>(replacement);
+
+    replace_nulls_with_scalar<<<grid.num_blocks, BLOCK_SIZE, 0, stream>>>(nrows,
+                                                                          input.data<col_type>(),
+                                                                          input.null_mask(),
+                                                                          s1->data(),
+                                                                          output.data<col_type>());
+  }
+};
+
+template<>
+void replace_nulls_scalar_kernel_forwarder::operator ()<cudf::string_view>(cudf::column_view const& input,
+                                                                           cudf::scalar const* replacement,
+                                                                           cudf::mutable_column_view& output,
+                                                                           cudaStream_t stream,
+                                                                           rmm::mr::device_memory_resource* mr) {
+  CUDF_FAIL("Strings not supported for replacement");
+}
+
+
+} //end anonymous namespace
+
+
+namespace cudf {
+namespace detail {
+
+std::unique_ptr<cudf::column> replace_nulls(cudf::column_view const& input,
+                                            cudf::column_view const& replacement,
+                                            rmm::mr::device_memory_resource* mr,
+                                            cudaStream_t stream)
+{
+  if (input.size() == 0) {
+    return std::make_unique<cudf::column>(input);
+  }
+
+  CUDF_EXPECTS(nullptr != input.data<int32_t>(), "Null input data");
+
+  if (input.nullable() == false || input.null_count() == 0) {
+    return std::make_unique<cudf::column>(input);
+  }
+
+  CUDF_EXPECTS(input.type() == replacement.type(), "Data type mismatch");
+  CUDF_EXPECTS(replacement.size() == 1 || replacement.size() == input.size(), "Column size mismatch");
+  CUDF_EXPECTS(nullptr != replacement.data<int32_t>(), "Null replacement data");
+  CUDF_EXPECTS(replacement.nullable() == false || 0 == replacement.null_count(),
+               "Invalid replacement data");
+
+  std::unique_ptr<cudf::column> output = make_numeric_column(input.type(),
```

Comment:
Not correct, doesn't handle other fixed-width column types.

Reply:
Corrected.